### PR TITLE
Move Timestamp class from Firestore to Firebase Core

### PR DESCRIPTION
**Pull request by cherylEnkidu**
Move `Timestamp` class into `FirebaseCore`. `FirebaseFirestore.Timestamp` was changed to `FirebaseCore.Timestamp`.

### CODE REVIEW

**Review by paulb777 — APPROVED** [review]
Good to merge to main after July 8th.

It can merge to the release-11.0 branch before that.

**Comment by cherylEnkidu** [comment]
> Good to merge to main after July 8th.
> 
> It can merge to the release-11.0 branch before that.

Hi @paulb777 , just want to double check, I can merge to `main` branch after July 8th?

**Comment by paulb777** [comment]
Yes - 10.29.0 is releasing and the release-11.0 branch has merged to main.

It looks like there are some conflicts. Let icore know if you have any questions about them.

**Review by wu-hui — APPROVED** [review]

File: FirebaseCore/CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,8 @@
   XCFrameworks.
 - Update underlying FIRLogger implementation from `asl` to `os_log`.
 - Remove `FIRLoggerForceSTDERR` configuration option.
+- [changed] Move `Timestamp` class into `FirebaseCore`. `FirebaseFirestore.Timestamp`
+  was changed to `FirebaseCore.Timestamp`. (#13221)
```
Comment (ncooke3) [inline FirebaseCore/CHANGELOG.md:21]: > `FirebaseFirestore.Timestamp` was changed to `FirebaseCore.Timestamp`.

@paulb777 – This is a small break, but I believe we could add:
```swift
@_exported import class FirebaseCore.Timestamp
```
to selectively import and re-export that class so it stills appears under the `FirebaseFirestore` namespace (like what was done with Swift extensions post-consolidation)

Comment (paulb777) [reply FirebaseCore/CHANGELOG.md:21]: SGTM. It should minimize the break.

Comment (cherylEnkidu) [reply FirebaseCore/CHANGELOG.md:21]: Should I create a new PR to add this line in `FirebaseFirestoreSwift.swift` or there is a pr working in progress?

Comment (ncooke3) [reply FirebaseCore/CHANGELOG.md:21]: @cherylEnkidu, sure! I haven't started anything. For all the files in this PR where you needed to add `import FirebaseCore`, I _think_ you should be able to just replace those with `@_exported import class FirebaseCore.Timestamp`.

Comment (cherylEnkidu) [reply FirebaseCore/CHANGELOG.md:21]: Got it, I will send you a PR shortly.

**Pull request merged**